WalletWasabi.Fluent/ViewModels/Navigation/NavigationExtensions.cs

```diff
@@ -0,0 +1,55 @@
+using System.Threading.Tasks;
+using WalletWasabi.Fluent.ViewModels.Dialogs;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+
+namespace WalletWasabi.Fluent.ViewModels.Navigation;
+
+public static class NavigationExtensions
+{
+	public static void To<T>(this INavigate navigate, T viewModel, NavigationTarget target = NavigationTarget.Default, NavigationMode mode = NavigationMode.Normal) where T : RoutableViewModel
+	{
+		target = GetTarget(viewModel, target);
+		navigate.Navigate(target).To(viewModel, mode);
+	}
```

Comment:
passing `NavigationTarget` is just for saving some time in some cases?
So I can do
```c#
navigate.To(page, target);
```
Instead of
```c#
navigate.Navigate(target).To(page);
```
?

Reply:
@soosr passing `NavigationTarget` allows you to override the default and current one.

The logic goes as follows:

 - If you're "standing" on a `RoutableViewModel`, you will have a `currentTarget` which is the target of the ViewModel you're standing on.
 - Then there's a `DefaultTarget` which is the default target of the ViewModel you want to navigate to. This is defined in the metadata which is usually set by `[NavigationMetadata]` attribute.
 - Then there's a fallback mechanism where if `CurrentTarget == NavigationTarget.Default` the `DefaultTarget` value will be used instead.
All of the above can be overriden by explicitly specifying `NavigationTarget` in the method call. Source-generated fluent API uses this as well to allow user-defined target in fluent API calls, such as:

    Navigate().To().UnusedReceiveAddresses(...args, NavigationTarget.DialogScreen);

and the like